packages/mermaid/src/mermaidAPI.ts

```diff
@@ -374,219 +378,17 @@ export const removeExistingElements = (
-    ? diag.renderer.getClasses(text, diag)
-    : {};
-
-  const rules = createUserStyles(
-    config,
-    graphType,
-    // @ts-ignore convert renderer to TS.
-    diagramClassDefs,
-    idSelector
-  );
-
-  // svg is a HTML element (not a d3 node)
-  const style1 = document.createElement('style');
-  style1.innerHTML = rules;
-  svg.insertBefore(style1, firstChild);
-
-  // -------------------------------------------------------------------------------
-  // Draw the diagram with the renderer
-  try {
-    diag.renderer.draw(text, id, pkg.version, diag);
-  } catch (e) {
-    errorRenderer.draw(text, id, pkg.version);
-    throw e;
-  }
-
-  // This is the d3 node for the svg element
-  const svgNode = root.select(`${enclosingDivID_selector} svg`);
-  const a11yTitle = diag.db.getAccTitle?.();
-  const a11yDescr = diag.db.getAccDescription?.();
-  addA11yInfo(graphType, svgNode, a11yTitle, a11yDescr);
-
-  // -------------------------------------------------------------------------------
-  // Clean up SVG code
-  root.select(`[id="${id}"]`).selectAll('foreignobject > *').attr('xmlns', XMLNS_XHTML_STD);
-
-  // Fix for when the base tag is used
-  let svgCode = root.select(enclosingDivID_selector).node().innerHTML;
-
-  log.debug('config.arrowMarkerAbsolute', config.arrowMarkerAbsolute);
-  svgCode = cleanUpSvgCode(svgCode, isSandboxed, evaluate(config.arrowMarkerAbsolute));
-
-  if (isSandboxed) {
-    const svgEl = root.select(enclosingDivID_selector + ' svg').node();
-    svgCode = putIntoIFrame(svgCode, svgEl);
-  } else if (!isLooseSecurityLevel) {
-    // Sanitize the svgCode using DOMPurify
-    svgCode = DOMPurify.sanitize(svgCode, {
-      ADD_TAGS: DOMPURIFY_TAGS,
-      ADD_ATTR: DOMPURIFY_ATTR,
-    });
-  }
-
-  // -------------------------------------------------------------------------------
-  // Do any callbacks (cb = callback)
-  if (cb !== undefined) {
-    switch (graphType) {
-      case 'flowchart':
-      case 'flowchart-v2':
-        cb(svgCode, flowDb.bindFunctions);
-        break;
-      case 'gantt':
-        cb(svgCode, ganttDb.bindFunctions);
-        break;
-      case 'class':
-      case 'classDiagram':
-        cb(svgCode, classDb.bindFunctions);
-        break;
-      default:
-        cb(svgCode);
-    }
-  } else {
-    log.debug('CB = undefined!');
-  }
-  attachFunctions();
-
-  // -------------------------------------------------------------------------------
-  // Remove the temporary element if appropriate
-  const tmpElementSelector = isSandboxed ? iFrameID_selector : enclosingDivID_selector;
-  const node = select(tmpElementSelector).node();
-  if (node && 'remove' in node) {
-    node.remove();
-  }
-
-  if (parseEncounteredException) {
-    throw parseEncounteredException;
-  }
-
-  return svgCode;
-};
-
-/**
- * @deprecated This is an internal function and should not be used. Will be removed in v10.
- */
+export interface RenderResult {
```

Comment:
Maybe move this declaration to top of file?

Reply:
Done